src/collective/taxonomy/jsonimpl.py

```diff
@@ -71,19 +67,8 @@ def get_data(self):
 
     def get_languages_mapping(self):
         """Get mapping token/value for languages."""
-        portal = api.portal.get()
-        portal_state = queryMultiAdapter(
-            (portal, portal.REQUEST), name=u'plone_portal_state')
-        mapping = portal_state.locale().displayNames.languages
         language_tool = api.portal.get_tool('portal_languages')
-        supported_langs = language_tool.supported_langs
-        languages_mapping = {
-            get_lang_code(lang): mapping[get_lang_code(lang)].capitalize()
-            for lang in supported_langs
-        }
-        # add taxonomy's default language if it is not in supported langs
-        default_lang = self.taxonomy.default_language
-        languages_mapping[default_lang] = mapping[default_lang].capitalize()
+        languages_mapping = dict(language_tool.listAvailableLanguages())
```

Comment:
erm ... you've remove the code which adds eventually existing taxonomies in no more existing portal languages ... I think we should still provide this *hidden* taxonomies in edit view

Reply:
by reading the code could not understand what was going on.

will try to change variable names to make it clear and add a test

Reply:
by simplifying that way, you get all the available languages on the edit view, not just the supported ones in the language settings ... this makes it hard to translate taxonomies in the correct supported language ... I'd recommend reverting the last commit ...

Reply:
done